pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -333,17 +363,53 @@ func (d *DependenciesDistributor) findOrphanAttachedBindings(independentBinding
 		return nil, err
 	}
 
-	dependenciesSets := sets.NewString()
-	for _, dependency := range dependencies {
-		key := generateDependencyKey(dependency.Kind, dependency.APIVersion, dependency.Namespace, dependency.Name)
-		dependenciesSets.Insert(key)
+	dependenciesMaps := make(map[string][]int, 0)
+	for index, dependency := range dependencies {
+		key := generateDependencyKey(dependency.Kind, dependency.APIVersion, dependency.Namespace)
+		dependenciesMaps[key] = append(dependenciesMaps[key], index)
 	}
 
 	var orphanAttachedBindings []*workv1alpha2.ResourceBinding
 	for _, attachedBinding := range attachedBindings {
-		key := generateDependencyKey(attachedBinding.Spec.Resource.Kind, attachedBinding.Spec.Resource.APIVersion, attachedBinding.Spec.Resource.Namespace, attachedBinding.Spec.Resource.Name)
-		if !dependenciesSets.Has(key) {
+		key := generateDependencyKey(attachedBinding.Spec.Resource.Kind, attachedBinding.Spec.Resource.APIVersion, attachedBinding.Spec.Resource.Namespace)
+		indexes, exist := dependenciesMaps[key]
+		if !exist {
 			orphanAttachedBindings = append(orphanAttachedBindings, attachedBinding)
+			continue
+		}
+		for _, idx := range indexes {
+			dependency := dependencies[idx]
+			if len(dependency.Name) != 0 {
+				if dependency.Name != attachedBinding.Spec.Resource.Name {
+					orphanAttachedBindings = append(orphanAttachedBindings, attachedBinding)
```

Comment:
There may be an issue with the handling here. I suggest extracting the processing of the entire for loop into a function and adding tests to ensure its proper functioning.